app/src/main/java/com/example/pix/home/fragments/ComposeFragment.java

```diff
@@ -1,42 +1,219 @@
 package com.example.pix.home.fragments;
 
+import android.annotation.SuppressLint;
+import android.app.Activity;
+import android.content.Context;
+import android.content.pm.PackageManager;
+import android.hardware.Camera;
 import android.os.Bundle;
 
+import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+import android.view.GestureDetector;
+import android.view.Gravity;
 import android.view.LayoutInflater;
+import android.view.MotionEvent;
+import android.view.PixelCopy;
 import android.view.View;
 import android.view.ViewGroup;
+import android.widget.Button;
 import android.widget.FrameLayout;
-import com.google.ar.core.Config;
-import com.google.ar.core.Session;
-import com.google.ar.sceneform.ux.ArFragment;
+import android.widget.ImageView;
+import android.widget.LinearLayout;
+import android.widget.PopupWindow;
+import android.widget.SearchView;
+import android.widget.Toast;
+
+import com.example.pix.R;
+import com.example.pix.chat.ChatActivity;
+import com.example.pix.chat.ChatFragment;
+import com.example.pix.home.activities.HomeActivity;
+import com.example.pix.home.adapters.SearchAdapter;
+import com.example.pix.home.utils.CameraPreview;
+import com.parse.ParseException;
+import com.parse.ParseFile;
+import com.parse.ParseQuery;
+import com.parse.ParseUser;
+import com.parse.SaveCallback;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ComposeFragment extends Fragment {
 
-import java.util.EnumSet;
-import java.util.Set;
+    ChatActivity mActivity;
+    public static ParseFile image;
 
-public class ComposeFragment extends ArFragment {
+    @Override
+    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        return inflater.inflate(R.layout.fragment_compose, container, false);
+    }
 
     @Override
-    protected Config getSessionConfiguration(Session session) {
-        Config config = new Config(session);
-        config.setAugmentedFaceMode(Config.AugmentedFaceMode.MESH3D);
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        if (!checkCameraHardware(getContext())) {
+            Toast.makeText(getContext(), "No camera available!", Toast.LENGTH_SHORT).show();
+            System.exit(0);
+        }
+    }
+
+    final int[] currCamera = new int[1];
+    final Camera[] c = new Camera[1];
+    final CameraPreview[] preview = new CameraPreview[1];
```

Comment:
Why do you need to make them arrays of size 1?

Reply:
These are those values that I need to declare as a final single element array. I needed to make them global values, and so I declared the actually Arrays as a single element final array.

Reply:
I don't think so. These are members of a class, in your anonymous class, you can assign to them as you want. They don't need to be final.

Reply:
Interesting, I didn't know this. I'll have to look up the difference, thanks!